Sources/MapboxDirections/Incident.swift

```diff
@@ -21,46 +20,144 @@ public struct Incident: Codable, Equatable {
         case geometryIndexStart = "geometry_index_start"
         case geometryIndexEnd = "geometry_index_end"
     }
-
-    var identifier: String
-    var type: String
-    var description: String
-    var creationTime: String
-    var startTime: String
-    var endTime: String
-    var impact: String?
-    var subtype: String?
-    var subtypeDescription: String?
-    var alertCodes: [Int]
-    var lanesBlocked: [Int]
-    var geometryIndexStart: Int
-    var geometryIndexEnd: Int
+    
+    /// Defines known types of incidents.
+    ///
+    /// Each incident may or may not have specific set of data, depending on it's `type`
+    public enum IncidentType: String {
+        case Accident = "accident"
+        case Congestion = "congestion"
+        case Construction = "construction"
+        case DisabledVehicle = "disabled_vehicle"
+        case LaneRestriction = "lane_restriction"
+        case MassTransit = "mass_transit"
+        case Miscellaneous = "miscellaneous"
+        case OtherNews = "other_news"
+        case PlannedEvent = "planned_event"
+        case RoadClosure = "road_closure"
+        case RoadHazard = "road_hazard"
+        case Weather = "weather"
+        
+        case Unknown = "<<unknown type>>"
+    }
+    
+    /// Incident identifier
+    public var identifier: String
+    /// The kind of an incident
+    public var type: IncidentType
+    var rawType: String
+    /// Short description of an incident. May be used as an additional info.
+    public var description: String
+    /// Date when incident item was created. Uses ISO8601 format
+    public var creationDate: Date
+    /// Date when incident happened. Uses ISO8601 format
+    public var startDate: Date
+    /// Date when incident shall end. Uses ISO8601 format
+    public var endDate: Date
```

Comment:
Now that these properties are of type Date, they don’t follow any string format.

Reply:
Removed